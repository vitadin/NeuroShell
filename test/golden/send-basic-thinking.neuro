%% Basic send command test with thinking blocks
%% Tests the core user workflow: just use \send and everything works
%% This verifies auto-session creation, auto-model creation, and thinking blocks

\set _style dark1

\echo === Basic Send with Thinking Blocks Test ===
\echo Testing seamless \send workflow with mock LLM thinking blocks

%% Use \send directly - should auto-create session and model
\echo Testing first message...
\send Hello, can you help me with a simple question?

\echo Testing follow-up message...
\send What's the weather like today?

\echo Testing third message to verify conversation flow...
\send Thanks for your help!

%% Show the results
\echo Session created by \send command:
\session-show

\echo Active model created by \send command:
\get[#active_model_name]
\echo Model: ${#active_model_name}