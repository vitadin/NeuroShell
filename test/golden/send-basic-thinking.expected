Setting _style = dark1
=== Basic Send with Thinking Blocks Test ===
Testing seamless \send workflow with mock LLM thinking blocks
Testing first message...
  --- OpenAI reasoning trace: ---                                             
  Thinking about the user's message: "Hello, can you help me with a simple    
  question?". This helps verify the message flow in tests. The user sent 1    
  messages total, and I need to provide a helpful response.                   
  --- End thinking ---                                                        
  This is a mocking reply (received 1 messages, last: Hello, can you help me  
  with a simple question?)                                                    
Testing follow-up message...
  --- OpenAI reasoning trace: ---                                             
  Thinking about the user's message: "What's the weather like today?". This   
  helps verify the message flow in tests. The user sent 3 messages total, and 
  I need to provide a helpful response.                                       
  --- End thinking ---                                                        
  This is a mocking reply (received 3 messages, last: What's the weather like 
  today?)                                                                     
Testing third message to verify conversation flow...
  --- OpenAI reasoning trace: ---                                             
  Thinking about the user's message: "Thanks for your help!". This helps      
  verify the message flow in tests. The user sent 5 messages total, and I need
  to provide a helpful response.                                              
  --- End thinking ---                                                        
  This is a mocking reply (received 5 messages, last: Thanks for your help!)  
Session created by \send command:
Session: Session 1 (ID: 00000001)
System: You are a helpful assistant.
Created: 2025-01-01 00:00:01
Updated: 2025-01-01 00:00:14
Messages: 6 total
[1] user (00:00:02): Hello, can you help me with a simple question?
[2] assistant (00:00:05): --- OpenAI reasoning trace: ---\n\n  Thi... you help me with a simple question?) (361 chars)
[3] user (00:00:07): What's the weather like today?
[4] assistant (00:00:09): --- OpenAI reasoning trace: ---\n\n  Thi...last: What's the weather like today?) (329 chars)
[5] user (00:00:11): Thanks for your help!
[6] assistant (00:00:13): --- OpenAI reasoning trace: ---\n\n  Thi...essages, last: Thanks for your help!) (311 chars)
Active model created by \send command:
#active_model_name = default_model
Model: default_model